moveit_ros/planning/rdf_loader/test/test_rdf_integration.cpp

```diff
@@ -66,6 +66,28 @@ TEST(RDFIntegration, topic_based)
   EXPECT_EQ("gonzo", loader.getSRDF()->getName());
 }
 
+TEST(RDFIntegration, executor)
+{
+  rclcpp::Node::SharedPtr node = std::make_shared<rclcpp::Node>("executor");
+
+  // Create a thread to spin an Executor.
+  std::promise<void> promise;
+  auto shared_future = promise.get_future().share();
+  std::thread([node, shared_future]() {
+    rclcpp::executors::SingleThreadedExecutor executor;
+    executor.add_node(node);
+    executor.spin_until_future_complete(shared_future, std::chrono::seconds(10));
+  }).detach();
```

Comment:
Is there a reason you detach this thread instead of joining with it after you set the value of the promise?  Also, you can move the initialization of the shared future into the capture if you'd like as you don't need it outside of the lambda.

```suggestion
  std::thread([node, shared_future = promise.get_future().share()]() {
    rclcpp::executors::SingleThreadedExecutor executor;
    executor.add_node(node);
    executor.spin_until_future_complete(shared_future, std::chrono::seconds(10));
  }).detach();
```

Reply:
After reading more about it, using detach is just fine.  There really is no difference and using it here simplifies the test which is nice.